Python/errors.c

```diff
@@ -1850,6 +1850,54 @@ PyErr_SyntaxLocationEx(const char *filename, int lineno, int col_offset)
     Py_XDECREF(fileobj);
 }
 
+/* Raises a SyntaxError and returns -1.
+ * If something goes wrong, a different exception may be raised.
+ */
+int
+_PyErr_RaiseSyntaxError(PyObject *msg, PyObject *filename, int lineno, int col_offset,
+                        int end_lineno, int end_col_offset)
+{
+    PyObject *loc_obj = PyErr_ProgramTextObject(filename, lineno);
+    if (loc_obj == NULL) {
+        loc_obj = Py_None;
```

Comment:
Unbalanced reference counting. Increment the refcount.

Reply:
Py_None is immutable.

Reply:
But we're not increasing Py_None in other places, so it wouldn't be consistent.

Reply:
We should.

Reply:
Can you bring this up with the c api workgroup?

(My view is that we shouldn't try to enforce something we can't test, because that is doomed to fail).

Reply:
I added it following discussion on discord. It's easier to take out than put in, so let it be so for now.

Reply:
Out of curiosity, what's the benefit of `INCREF`'ing known immutable objects? It seems like it would just hurt performance.